lib/ace/transport_app.rb

```diff
@@ -214,44 +214,46 @@ def self.trusted_facts(certname)
         validate_schema(@schemas["run_task"], body)
         opts = body['target'].merge('protocol' => 'remote')
 
-        # This is a workaround for Bolt due to the way it expects to receive the target info
-        # see: https://github.com/puppetlabs/bolt/pull/915#discussion_r268280535
-        # Systems calling into ACE will need to determine the nodename/certname and pass this as `name`
-        target = [Bolt::Target.new(body['target']['host'] || body['target']['name'], opts)]
+        inventory = Bolt::Inventory.empty
+        target_data = {
+          'name' => body['target']['host'] || body['target']['name'] || 'remote',
+          'config' => {
+            'transport' => 'remote'
+          }
+        }
+        target = [Bolt::Target.from_hash(target_data, inventory)]
       rescue ACE::Error => e
         request_error = {
```

Comment:
I'm wondering if we can just construct Bolt::Results here. https://github.com/puppetlabs/bolt/blob/master/lib/bolt/result.rb#L10 that way we dont have to keep trying to mimic the structure or a proper result.

Reply:
Nope. Any error here and there will not yet be a Target. See https://github.com/puppetlabs/bolt/pull/1693 for parity w/bolt-server